hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/MetricsHook.java

```diff
@@ -44,24 +44,13 @@ public class MetricsHook implements Hook {
     private final LongCounter evaluationErrorCounter;
     private final List<DimensionDescription> dimensionDescriptions;
     private final Function<ImmutableMetadata, Attributes> extractor;
+    private final Attributes extraDimensions;
 
     /**
      * Construct a metric hook by providing an {@link OpenTelemetry} instance.
      */
     public MetricsHook(final OpenTelemetry openTelemetry) {
-        this(openTelemetry, Collections.emptyList());
-    }
-
-    /**
-     * Construct a metric hook with {@link OpenTelemetry} instance and a list of {@link DimensionDescription}.
-     * Provided dimensions are attempted to be extracted from ImmutableMetadata attached to
-     * {@link FlagEvaluationDetails}.
-     *
-     * @deprecated - This constructor is deprecated. Please use {@link MetricHookOptions} based options
-     */
-    @Deprecated
-    public MetricsHook(final OpenTelemetry openTelemetry, final List<DimensionDescription> dimensions) {
-        this(openTelemetry, MetricHookOptions.builder().setDimensions(dimensions).build());
```

Comment:
Good idea to remove this, but I think this makes this a breaking change (which is fine IMO)